### PR TITLE
rgw/admin: Add support for user bucket quota

### PR DESCRIPTION
**Pull request by anoopcs9**
Following APIs have been added to set and get bucket quotas for users.
- _SetBucketQuota_
- _GetBucketQuota_

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code
- [x] Is this a new API? Added a new file that begins with `//go:build ceph_preview`
- [x] Ran `make api-update` to record new APIs

Credits: [Pavel Boev](https://github.com/R4scal)

### CODE REVIEW

**Review by phlogistonjohn — COMMENTED** [review]
Looks OK to me, but RGW is my weakest area. I'll hold off approving so that both thotz and ansiwen can get a chance to review before merging. But if you want my approval at some point, just ping me.

**Review by thotz — DISMISSED** [review]

**Comment by anoopcs9** [comment]
> But if you want my approval at some point, just ping me.

@phlogistonjohn Now if you may.. 😐

**Review by thotz — APPROVED** [review]

**Review by phlogistonjohn — APPROVED** [review]

**Pull request merged**
